### ISSUE TITLE
Drive activity leds like AE1? 

### ISSUE THREAD

**Issue by aadje93** [AppliedEnergistics/Applied-Energistics-2#7439]
### Describe the feature

in AE 1 we had the B:gfxDrivesHighDef=true / false option to either enable or disable the drive activity leds. I realy liked those flashing leds in the disk racks making it look like servers in a tower with the controller being my "mainframe"

In the AE2 client config and server config (playing ATM8) i couldn't find this option anymore, can we still enable this? Or is it completely removed due to performance hits? 

I'm playing on a pretty beefy computer + server so i would realy like those leds flashing again :)

### Reasons why it should be considered

Immersion, was a great option on AE1

### Additional details

in AE1 we had the config option B:gfxDrivesHighDef;

[](https://appliedenergistics.org/ae1-site-archive/Config-Options/index.html)

**Comment by aadje93** [comment]
Yes  on AE1 the drives dod blink om activity, was pretty fun bit did cause some lagg on low spec server/cliënt thats why there was a config option to disable it. But on AE 2 it seems completely removed?

**Comment by mindforger** [comment]
this option had a bad reputation for being an FPS killer in the past the way it was implemented, but practically yes the LED blinked and so did the cables

currently the LEDs indicate unpowered(dark/grey), free(green), item types full(yellow) or completely full (red)

maybe add a blue flash whenever items are moved in/out of the cell

but without knowing the implications to sync access and rendering, this sounds like it has to be either imprecise(mening it does not indicate each an every update) or may have an significant impact on the rendering performance


something like an additional but only temporary internal state that is set during tick when something has changed inside the drive and causes the rendering to update the LED for a certain time with blue for 100ms and then for at least 500ms show the actual state to not signalling permanently blue on an very active drive and limit the syncing with the simulation (this option had a bad reputation in the past for being FPS killing)

OR

make an activity counter that indicates the number of accesses over a cerrtain time (as an moving average) and in rendering update the LED in some random pattern that is more flickery the higher the counter gets to decouple it even more

**Comment by aadje93** [comment]
For me it 2as nice to see activity, so a average over time to prevent fps killing would be very nice.

**Comment by Sea-Kerman** [comment]
> currently the LEDs indicate unpowered(dark/grey), free(green), item types full(yellow) or completely full (red)
> 
and blue for “some items and types, but not full on either”

**Comment by aadje93** [comment]
yeah i know the current state, but my request/idea was bringing back the activity blinking from AE1 ;), doesnt have to be in sync. But just like a network port blinking like 1x/s if there "is" activity in the past 2 secs or so is a nice adition, and if i calculate correctly it shouldn't be more then 1/40ticks caclulation. (note bytes stored in disk, 40 ticks later check difference, difference >1 blink for 2x or so every 20 ticks)

**Comment by mindforger** [comment]
> > currently the LEDs indicate unpowered(dark/grey), free(green), item types full(yellow) or completely full (red)
> 
> and blue for “some items and types, but not full on either”

well it's light blue, activity could be a more solid blue then, but yeah i totally forgot :D

**Comment by aadje93** [comment]
off - on would be the best :). thats how hdd activity leds on servers work too (could show you an exampe of my 24 disk Truenas machines  *also having blue leds*

**Comment by mindforger** [comment]
Going off on might be misleading and mistaken for power issues imho

**Comment by Shaggy84675** [comment]
+1

I miss that feature too from AE1
